node/src/actors/chain_manager/mining.rs

```diff
@@ -435,22 +479,22 @@ fn build_block(
         tally_txns,
     };
 
+    // Put an invalid signature in the block, because signing
```

Comment:
Why not signing yet? Why no `TODO`? __WHY?__

Reply:
Because now we need to sign the block after building it, so the `build_block` function would need access to the signature manager actor. Therefore I will change the function signature to return `(BlockHeader, BlockTransactions)`.